modules/reindex/src/main/java/org/elasticsearch/reindex/RestReindexAction.java

```diff
@@ -68,6 +69,12 @@ protected ReindexRequest buildRequest(RestRequest request, NamedWriteableRegistr
             internal.setRequireAlias(request.paramAsBoolean(DocWriteRequest.REQUIRE_ALIAS, false));
         }
 
+        // Do not open scroll if limit <= scroll size
+        var docsPerScroll = internal.getSearchRequest().source().size();
+        if (internal.getMaxDocs() != -1 && internal.getMaxDocs() <= docsPerScroll && internal.isAbortOnVersionConflict()) {
+            internal.getSearchRequest().scroll((Scroll) null);
+        }
```

Comment:
Would it be possible to do this in the transport layer instead? That way it would apply also for internal reindex usages. We try to keep the REST layer free of logic.

If that is much more difficult, I do not mind it living here though.